### PR TITLE
Automating Dependabot with GitHub Actions

### PR DESCRIPTION
**Pull request by utam0k**
https://docs.github.com/en/code-security/dependabot/working-with-dependabot/automating-dependabot-with-github-actions#enable-auto-merge-on-a-pull-request

### CODE REVIEW

File: .github/workflows/dependabot_auto.yaml [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    if: ${{ github.actor == 'dependabot[bot]' }}
+    steps:
+      - name: Dependabot metadata
+        id: metadata
+        uses: dependabot/fetch-metadata@v1.3.5
+        with:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+      - name: Approve & enable auto-merge for Dependabot PR
+        if: |
+          steps.metadata.outputs.update-type == 'version-update:semver-patch'
```
Comment (utam0k) [inline .github/workflows/dependabot_auto.yaml:25]: Only updating patch version, this workflow began to run.

File: .github/workflows/dependabot_auto.yaml [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+        run: |
+          gh pr review --approve "$PR_URL"
+          gh pr edit "$PR_URL" -t "(auto merged) $PR_TITLE"
+        env:
+          PR_URL: ${{ github.event.pull_request.html_url }}
+          PR_TITLE: ${{ github.event.pull_request.title }}
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+      - name: Automerge
+        id: automerge
+        uses: pascalgn/automerge-action@v0.15.5
```
Comment (YJDoc2) [inline .github/workflows/dependabot_auto.yaml:35]: Should we use `gh pr merge --auto --merge "$PR_URL"` as written in the link you have mentioned? The automerge action readme states :
>  This functionality is now available directly in GitHub as [auto-merge](https://docs.github.com/en/github/collaborating-with-issues-and-pull-requests/automatically-merging-a-pull-request). Note that GitHub does not currently support auto-rebasing pull requests. The automerge-action project will still be maintained, but users are encouraged to switch to auto-merge for simple workflows, as it offers a faster and more stable experience.

WDYT @utam0k ?

Comment (utam0k) [reply .github/workflows/dependabot_auto.yaml:35]: @YJDoc2 Good point. Auto merge affects the entire repository.
https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/configuring-pull-request-merges/managing-auto-merge-for-pull-requests-in-your-repository

But I don't think we have CI, unit tests, etc... enough to enable this feature yet.

Comment (YJDoc2) [reply .github/workflows/dependabot_auto.yaml:35]: I was talking about using gh cli to merge the PR as I have mentioned in the comment, but sure the actions also seems pretty decent.
Also I think we have enough tests and CI checks? correct me if I'm wrong.
I'm approving the PR, as it looks good to me, but not merging ; you can merge it after you see this comment :+1:

**Review by YJDoc2 — APPROVED** [review]
lgtm :+1:

File: .github/workflows/dependabot_auto.yaml (hunk 2 above)
Comment (utam0k) [reply .github/workflows/dependabot_auto.yaml:35]: > Also I think we have enough tests and CI checks? correct me if I'm wrong.

This is debatable. For example, I feel that the wasm and checkpoint features have not been adequately tested with CI. I'd like to get your opinition and if we need, let's create an issue and address it. You opinion is always great and respects it.

Comment (YJDoc2) [reply .github/workflows/dependabot_auto.yaml:35]: That is true. We can create issues for that and deal with them. For now, this PR should do well :+1:
Thanks :)

**Pull request merged**